File: 1911-maximum-alternating-subsequence-sum/1911-maximum-alternating-subsequence-sum.py
class Solution:
    def maxAlternatingSum(self, nums: List[int]) -> int:
        
        return sum([max(0, nums[i+1] - nums[i]) for i in range(len(nums)-1)]) + nums[0]
        
#         ans = 0
#         for i in range(len(nums)-1):
#             ans += max(0, nums[i+1] - nums[i])

#         return ans + nums[0]

File: 1911-maximum-alternating-subsequence-sum/1911-maximum-alternating-subsequence-sum.java
class Solution {
    public long maxAlternatingSum(int[] nums) {
        long ans = 0;
        for(int i = 0; i < nums.length -1; i++){
            ans += Math.max(0, nums[i+1] - nums[i]);
        }
        return ans + nums[0];
    }
}